### PR TITLE
DDF-2383 Upgraded to CXF 3.1.7

### PR DESCRIPTION
**Pull request by shaundmorris**
#### What does this PR do?
Upgrades the CXF version to 3.1.7
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@ahoffer @AzGoalie @coyotesqrl @clockard 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire 
@stustison
#### How should this be tested?
Full build
Testing of endpoints (might consider basic performance testing)
Login/logout 
#### Any background context you want to provide?
The `TokenParameters.getWebServiceContext()` method was removed in-between CXF versions. 
There were new feature pre-reqs added so need review if these were placed appropriately. Currently added to the sts-prereqs feature in the security-services-app/features.xml
#### What are the relevant tickets?
DDF-2383
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

---
This change is [Reviewable](https://reviewable.io/reviews/codice/ddf/1135)

### CODE REVIEW

**Comment by kcwire** [comment]
👍

File: platform/security/sts/security-sts-server/src/main/resources/OSGI-INF/blueprint/cxf-sts.xml [hunk 1]
```diff
@@ -172,9 +172,7 @@
     </bean>
 
     <bean id="service"
-          class="ddf.security.sts.StsStaticService">
-        <cm:managed-properties persistent-id="ddf.security.sts.StsStaticService"
```
Comment (stustison) [inline platform/security/sts/security-sts-server/src/main/resources/OSGI-INF/blueprint/cxf-sts.xml]: Did we remove this for some reason? There is a metatype for it.

Comment (shaundmorris) [inline platform/security/sts/security-sts-server/src/main/resources/OSGI-INF/blueprint/cxf-sts.xml]: Oh yeah good catch. I was still testing the removal of the `StsStaticService`. I will follow-up with removing the Java class and all references. It was originally used as a workaround for https://issues.apache.org/jira/browse/CXF-6673

**Comment by stustison** [comment]
👍  as long as the static service stuff gets cleaned up

File: platform/security/sts/security-sts-server/src/main/resources/OSGI-INF/blueprint/cxf-sts.xml (hunk 1 above)
Comment (clockard) [inline platform/security/sts/security-sts-server/src/main/resources/OSGI-INF/blueprint/cxf-sts.xml]: Good to see that this is finally getting fixed.

**Comment by coyotesqrl** [comment]
@shaundmorris Where are we on this PR?

**Comment by shaundmorris** [comment]
@coyotesqrl apologies for the delay, it should be cleaned up now. @stustison please feel to take another look.

**Comment by stustison** [comment]
LGTM 👍

**Comment by shaundmorris** [comment]
I was investigating a failed itest on bamboo. All itests passed on the last bamboo run, but there was a java.lang.OutOfMemoryError after the build was successful.

**Comment by kcwire** [comment]
@shaundmorris any reason we shouldn't merge? - bamboo looks like a false negative

**Comment by shaundmorris** [comment]
I think we are good here just was waiting for any additional comments, but seems like I have received everything I was going to get.

**Pull request merged**